HeavyFlavorAnalysis/SpecificDecay/src/BPHBdToJPsiKxBuilder.cc

```diff
@@ -0,0 +1,226 @@
+// This Class' Header --
+//-----------------------
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHBdToJPsiKxBuilder.h"
+
+//-------------------------------
+// Collaborating Class Headers --
+//-------------------------------
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHRecoBuilder.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHPlusMinusCandidate.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHRecoCandidate.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHMassSelect.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHChi2Select.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHMassFitSelect.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHParticleMasses.h"
+
+//---------------
+// C++ Headers --
+//---------------
+using namespace std;
+
+//-------------------
+// Initializations --
+//-------------------
+
+
+//----------------
+// Constructors --
+//----------------
+BPHBdToJPsiKxBuilder::BPHBdToJPsiKxBuilder( const edm::EventSetup& es,
+    const std::vector<BPHPlusMinusConstCandPtr>& jpsiCollection,
+    const std::vector<BPHPlusMinusConstCandPtr>&  kx0Collection ):
+  jPsiName( "JPsi" ),
+   kx0Name(  "Kx0" ),
+  evSetup( &es ),
+  jCollection( &jpsiCollection ),
+  kCollection( & kx0Collection ) {
+  jpsiSel = new BPHMassSelect   ( 2.80, 3.40 );
+  mkx0Sel = new BPHMassSelect   ( 0.80, 1.00 );
+  massSel = new BPHMassSelect   ( 3.50, 8.00 );
+  chi2Sel = new BPHChi2Select   ( 0.02 );
+  mFitSel = new BPHMassFitSelect( jPsiName,
+                                  BPHParticleMasses::jPsiMass,
+                                  BPHParticleMasses::jPsiMWidth,
+                                  5.00, 6.00 );
+  massConstr = true;
+  minPDiff = 1.0e-4;
+  updated = false;
+}
+
+//--------------
+// Destructor --
+//--------------
+BPHBdToJPsiKxBuilder::~BPHBdToJPsiKxBuilder() {
+  delete jpsiSel;
+  delete mkx0Sel;
+  delete massSel;
+  delete chi2Sel;
+  delete mFitSel;
+}
+
+//--------------
+// Operations --
+//--------------
+vector<BPHRecoConstCandPtr> BPHBdToJPsiKxBuilder::build() {
+
+  if ( updated ) return bdList;
+
+  BPHRecoBuilder bBd( *evSetup );
+  bBd.setMinPDiffererence( minPDiff );
+  bBd.add( jPsiName, *jCollection );
+  bBd.add(  kx0Name, *kCollection );
+  bBd.filter( jPsiName, *jpsiSel );
+  bBd.filter(  kx0Name, *mkx0Sel );
+
+  bBd.filter( *massSel );
+  bBd.filter( *chi2Sel );
+  if ( massConstr ) bBd.filter( *mFitSel );
+
+  bdList = BPHRecoCandidate::build( bBd );
+//  int iBd;
+//  int nBd = ( massConstr ? bdList.size() : 0 );
+//  for ( iBd = 0; iBd < nBd; ++iBd ) {
+//    BPHRecoCandidate* cptr( const_cast<BPHRecoCandidate*>(
+//                            bdList[iBd].get() ) );
+//    BPHRecoConstCandPtr jpsi = cptr->getComp( jPsiName );
+//    double jMass = jpsi->constrMass();
+//    if ( jMass < 0 ) continue;
+//    double sigma = jpsi->constrSigma();
+//    cptr->kinematicTree( jPsiName, jMass, sigma );
+//  }
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant